File: src/services/auth.service.ts
import { api } from './api'
import type { AuthResponse, LoginRequest, RegisterRequest, User } from '../types/user.types'

export const authService = {
  login: (data: LoginRequest) =>
    api.post<AuthResponse>('/auth/login', data).then((r) => r.data),

  register: (data: RegisterRequest) =>
    api.post<AuthResponse>('/auth/register', data).then((r) => r.data),

  logout: () =>
    api.post('/auth/logout').catch(() => {}), 

  me: () =>
    api.get<User>('/auth/me').then((r) => r.data),

  updateProfile: (data: Partial<User>) =>
    api.put<User>('/auth/me', data).then((r) => r.data),
}

File: user-service/src/main/java/com/nwt/userservice/service/AuthService.java
package com.nwt.userservice.service;

import com.nwt.userservice.dto.request.LoginRequest;
import com.nwt.userservice.dto.request.RegisterRequest;
import com.nwt.userservice.dto.response.AuthResponse;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    AuthResponse refresh(String refreshToken);
}
